src/plugins/irc/irc-protocol.c

```diff
@@ -514,6 +514,98 @@ IRC_PROTOCOL_CALLBACK(cap)
                 irc_server_sendf (server, 0, NULL, "CAP END");
         }
     }
+    else if (strcmp (argv[3], "NEW") == 0)
+    {
+        if (argc > 4)
+        {
+            ptr_caps = (argv_eol[4][0] == ':') ? argv_eol[4] + 1 : argv_eol[4];
+            weechat_printf_date_tags (
+                server->buffer, date, NULL,
+                _("%s%s: client capability, now available: %s"),
+                weechat_prefix ("network"), IRC_PLUGIN_NAME, ptr_caps);
+
+            /* Assume that we're not requesting any already-enabled capabilities
+             * TODO SASL Reauthentication */
+            ptr_cap_option = IRC_SERVER_OPTION_STRING(server,
```

Comment:
This is all nearly copy-paste from 366-408. Why not make a cap parsing function?